auth/src/main/java/com/firebase/ui/auth/provider/FacebookProvider.java

```diff
@@ -182,9 +201,32 @@ private IdpResponse createIdpResponse(String email, LoginResult loginResult) {
                 loginResult.getAccessToken().getToken());
     }
 
-    private void onSuccess(String email, LoginResult loginResult) {
-        gcCallbackManager();
-        mCallbackObject.onSuccess(createIdpResponse(email, loginResult));
+    private void onSuccess(final String email, final LoginResult loginResult) {
+        if (sPublishPermissions.isEmpty()) {
+            gcCallbackManager();
+            mCallbackObject.onSuccess(createIdpResponse(email, loginResult));
+        } else {
+            LoginManager loginManager = LoginManager.getInstance();
+            loginManager.registerCallback(sCallbackManager, new FacebookCallback<LoginResult>() {
+                @Override
+                public void onSuccess(LoginResult result) {
+                    sPublishPermissions.clear();
+                    FacebookProvider.this.onSuccess(email, loginResult);
+                }
+
+                @Override
+                public void onCancel() {
+                    FacebookProvider.this.onCancel();
+                }
+
+                @Override
+                public void onError(FacebookException error) {
+                    FacebookProvider.this.onError(error);
+                }
+            });
+            loginManager.logInWithPublishPermissions((Activity) mCallbackObject,
```

Comment:
@samtstern Here's where I'm stuck. The code works ([see video](https://goo.gl/photos/t37UGVG4FiuQRTdx9)) for `AuthMethodPickerActivity` of course, but we can't do this anywhere else. This means we would have to make Facebook special and save the activity in some other way. (We can't save it from `startLogin` because the activity might die between then and when we get the callback.)

#### Some context

You were right, Facebook makes this harder than I originally thought. You can only request read permissions in the read method and write permissions in the write method. This raises the problem that we have to request a login twice which I feel is bad UX. In addition, keeping track of all write permissions is pain and I doubt anyone will be paying attention if Facebook updates their permission list.

#### My thoughts

The way it turned out is kinda gross and I feel like it's only going to get worse if we do it right. In addition, I think this feature extends beyond the scope of **Firebase**UI. The main point of this library is to connect various auth providers with a Firebase account. I feel like messing around with complex permissions that require approval from Facebook shouldn't really be our problem, especially since asking for those permissions happens after we've already accomplished our goal of creating a Firebase account. Given the above, I think we **shouldn't** support requesting publish permissions. @samtstern What are your thoughts on this?